atlp-rwanda/stackup2-barefoot-backend
# #171380780 rate an accommodation

### What does this PR ​ do​ ? 
- give rates to an accommodation you have traveled to

### Description of Task to be completed?
- create a rating model
- create handleRatingAccommodation function in `./src/services/accommodation.service.js file`
- create rateAccommodation function in `./src/controllers/accommodation.controller.js file`
- create rating Accommodation route in `./src/routes/api/accommodation.routes.js file`
- Have the following API route:
`POST /api/accommodations/rates`

### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- Make sure to rename .env.example to .env, 
- In .env file, provide all values for variables under `#Development database`  and `#General`
- Finally, run the following commands respectively:
`git fetch`
`git checkout ft-rate-accommodation-171380780`
`npm i`
`npm run migrations`
- If the above steps have been followed correctly, 
you should see a rating table created in your database
- Insert an accommodation in the database by running `npm run seed` 
- create a trip request 
- book accommodation for the trip request you created
- on `http://<your_server_hostname>:<server_port>/api/accommodations/rates` send
>"requestId":1
>"rates": 4
- on `http://<your_server_hostname>:<server_port>/api/accommodations/rates/:id` 
>provide the id of rated accommodation

## Any background context you want to provide?
- You must create an account and use the obtained token as an `authorization header`

### What are the relevant pivotal tracker stories?
[ #171380780](https://www.pivotaltracker.com/story/show/171380780)

### Screenshots 
![accommodationRated](https://user-images.githubusercontent.com/52660542/80721048-76717d80-8a99-11ea-9a18-71098859ccf8.PNG)

![didNotBook](https://user-images.githubusercontent.com/52660542/80721200-aae53980-8a99-11ea-91d3-d15193a3ef42.PNG)

![beforeDate](https://user-images.githubusercontent.com/52660542/80721288-c05a6380-8a99-11ea-8f6c-f99b626d20b5.PNG)

![invalidRate](https://user-images.githubusercontent.com/52660542/80721478-fb5c9700-8a99-11ea-9bb5-5d53ffe46a02.PNG)

![invalidRequestId](https://user-images.githubusercontent.com/52660542/80721526-0adbe000-8a9a-11ea-9b5d-401e326bef90.PNG)

![NotExistRequest](https://user-images.githubusercontent.com/52660542/80721594-1cbd8300-8a9a-11ea-8ced-bdcbb07fbf0d.PNG)

![notRatesOwner](https://user-images.githubusercontent.com/52660542/80721651-2fd05300-8a9a-11ea-99f8-4d62c429230f.PNG)

![ratedRequest](https://user-images.githubusercontent.com/52660542/80721729-44ace680-8a9a-11ea-8a93-d5d652b4f010.PNG)

### Questions:
N/A
